packages/brick_offline_first/lib/src/offline_queue/offline_rest_request_queue.dart

```diff
@@ -0,0 +1,42 @@
+import 'dart:async';
+
+import 'package:brick_offline_first/src/offline_queue/offline_queue_http_client.dart';
+import 'package:brick_offline_first/src/offline_queue/offline_request_queue.dart';
+import 'package:logging/logging.dart';
+import 'package:http/http.dart' as http;
+
+class OfflineRestRequestQueue extends OfflineRequestQueue<OfflineQueueHttpClient> {
+  /// The client responsible for resending requests
+  @override
+  // ignore: overridden_fields
+  final OfflineQueueHttpClient client;
+
+  OfflineRestRequestQueue({required this.client})
```

Comment:
All of this can be inherited from the super class
```suggestion
  OfflineRestRequestQueue({required OfflineQueueHttpClient client})
```